docs/strategies/python/poetry.md

```diff
@@ -0,0 +1,516 @@
+# Poetry
+
+[Poetry](https://python-poetry.org/) is a tool for dependency management and packaging in Python.
+
+## Project Discovery
+
+Find all files named `pyproject.toml` and `poetry.lock`. Pyproject must use poetry for the build system.
+
```

Comment:
They cannot per pep spec (it can only ref one python object). I've added poetry docs here for context. Also cleared ambiguity around what happens if the pyproject does not use poetry build backend (we do not discover this as poetry project).